### PR TITLE
fix(jest): support ESM setup files

### PR DESCRIPTION
**Pull request by sparten11740**
Add support for importing ESM files supplied in jest's `setupFiles` or `setupFilesAfterEnv`

### CODE REVIEW

File: tests/setup-files/setup.cjs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const path = require('path')
+
+global.SETUP_CJS = path.basename(__filename) // using some cjs stuff here to make sure it works
```
Comment (ChALkeR) [inline tests/setup-files/setup.cjs:3]: `globalThis` is the canonical name for the global object

File: tests/setup-files/setup.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import path from 'path'
+
+global.SETUP_JS_MODULE = path.basename(import.meta.url)
```
Comment (ChALkeR) [inline tests/setup-files/setup.js:3]: `import.meta.url` is an url, calling basename on it might be not ideal
does just `import.meta.filename` work here?

Comment (ChALkeR) [reply tests/setup-files/setup.js:3]: ah, this is actually testing that `import path` works, makes sense

Comment (sparten11740) [reply tests/setup-files/setup.js:3]: yeah, these 3 setup files are just for testing.

File: tests/setup-files/setup.cjs (hunk 1 above)
Comment (sparten11740) [reply tests/setup-files/setup.cjs]: replaced with [globalThis](https://github.com/ExodusMovement/test/pull/12/commits/1f6169bd3f4f352397069be4b1613a0a3ada4827)

**Review by ChALkeR — APPROVED** [review]
Let's land this and if anything needs to be fixed will change it later

**Pull request merged**